test/hotspot/jtreg/runtime/os/TestTracePageSizes.java

```diff
@@ -94,31 +112,97 @@
     // Store address ranges with known page size.
     private static LinkedList<RangeWithPageSize> ranges = new LinkedList<>();
     private static boolean debug;
+    private static int run;
 
-    // Parse /proc/self/smaps using a regexp capturing the address
-    // ranges, what page size they have and if they might use
-    // transparent huge pages. The pattern is not greedy and will
-    // match as little as possible so each "segment" in the file
-    // will generate a match.
+    private static long getKernelVersion() {
+        long kernelVersion = Platform.getOsVersionMajor() << 8 | Platform.getOsVersionMinor();
+        debug("kernelVersion " + Long.toHexString(kernelVersion));
+        return kernelVersion;
+    }
```

Comment:
Let's not do any bit magic? Just inline this to the only use and do the proper branchy "if"?

Reply:
+1